br/pkg/pdutil/pd_test.go

```diff
@@ -15,26 +15,18 @@ import (
 	"testing"
 
 	"github.com/coreos/go-semver/semver"
-	. "github.com/pingcap/check"
 	"github.com/pingcap/kvproto/pkg/metapb"
 	"github.com/pingcap/tidb/util/codec"
+	"github.com/stretchr/testify/require"
 	"github.com/tikv/pd/pkg/typeutil"
 	"github.com/tikv/pd/server/api"
 	"github.com/tikv/pd/server/core"
 	"github.com/tikv/pd/server/statistics"
 )
 
-func TestT(t *testing.T) {
-	TestingT(t)
-}
-
-type testPDControllerSuite struct {
-}
-
-var _ = Suite(&testPDControllerSuite{})
-
-func (s *testPDControllerSuite) TestScheduler(c *C) {
-	ctx := context.Background()
+func TestScheduler(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Can this test run in parallel?

Reply:
I tried but the test will fail sometimes. IMO it maybe related with `go func`, and we'd better to test serially.